sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.testing.metrics;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.metrics.data.Data;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableSummaryData;
+import io.opentelemetry.sdk.resources.Resource;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Immutable representation of all data collected by the {@link
+ * io.opentelemetry.sdk.metrics.data.MetricData} class.
+ */
+@Immutable
+@AutoValue
+public abstract class TestMetricData implements MetricData {
+  public static Builder builder() {
+    return new AutoValue_TestMetricData.Builder()
+        .setResource(Resource.empty())
+        .setName("name")
+        .setInstrumentationScopeInfo(InstrumentationScopeInfo.empty())
+        .setDescription("description")
+        .setUnit("1")
+        .setData(ImmutableSummaryData.empty())
+        .setType(MetricDataType.SUMMARY);
```

Comment:
Not sure setting data and type to arbitrary selections is better than omitting them.

Also, the value of this is limited without usable `Data` and `PointData` implementations.

@jkwatson WDYT about adding static create methods to all the interfaces in the [io.opentelemetry.sdk.metrics.data](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data)? These could return the [internal immutable implementations](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data), without requiring that we make all those classes public.

Reply:
hey @jack-berg it looks like all the immutable implementations already have public `create` methods. do you mind share a bit more detail on the last comment above? 

would it look better to just return the `AutoValue_TestMetricData.Builder()` form the `builder()` method?